### PR TITLE
Issue #1191 Avoid running ohmyzsh install more than once and correct the git URL

### LINKED ISSUE #1191: oh my zsh support broken and doesn't handle repeated provisioning
## Versions
- Vagrant: 2.2.4
- Provider: Virtualbox 5.2.26
- Homestead: 8.5.3

## Host operating system
macOS High Sierra 10.13.6

## Homestead.example.yaml
```yaml
ip: 192.168.10.10
memory: 2048
cpus: 4
provider: virtualbox
authorize: ~/.ssh/id_rsa.pub
keys:
    - ~/.ssh/id_rsa
folders:
    -
        map: .
        to: /home/vagrant/code
        type: rsync
        options:
            rsync__exclude: [".git/", ".idea", "web/sites/*/files", "web/sites/*/files-private", "web/artifacts", "web/sites/simpletest"]
            rsync__args: ["--verbose", "--archive", "--delete", "-z", "--chmod=g+rwX"]
sites:
    -
        map: mysite.local
        to: /home/vagrant/code/web
        type: apache
        php: '7.3'
        xhgui: 'true'
databases:
    - mysite
name: mysite
hostname: mysite.local
mongodb: true
ohmyzsh: true
variables:
    -
        key: ZSH_THEME
        value: robbyrussell
```

## Expected behavior

Should install oh my zsh as the shell with the robbyrussell theme.

## Actual behavior

I get errors on provision:

```
==> mysite: Running provisioner: shell...
    mysite: Running: script: Installing Oh-My-Zsh
    mysite: Cloning into '/home/vagrant/.oh-my-zsh'...
    mysite: fatal: unable to connect to github.com:
    mysite: github.com[0: 192.30.253.113]: errno=Connection timed out
    mysite: cp: cannot stat '/home/vagrant/.oh-my-zsh/templates/zshrc.zsh-template': No such file or directory
```
Also as I debugged and ran `vagrant provision` multiple times, I noticed that the script for installing oh my zsh doesn't check for a prior install and not all the commands are idempotent.

### PR DESCRIPTION
**Pull request by m4olivei**
Resolves #1191.

### CODE REVIEW

File: scripts/install-ohmyzsh.sh [hunk 1]
```diff
@@ -2,7 +2,13 @@
 
 # Install oh-my-zsh
 
-git clone git://github.com/robbyrussell/oh-my-zsh.git /home/vagrant/.oh-my-zsh
+if [ -f /home/vagrant/.oh-my-zsh ]
+then
+    echo "oh my zsh already installed."
+    exit 0
+fi
+
```
Comment (svpernova09) [inline scripts/install-ohmyzsh.sh]: Need to add a line here that touches the `/home/vagrant/.oh-my-zsh` like we do in other cases.

Comment (svpernova09) [reply scripts/install-ohmyzsh.sh]: Oh wait, no, we need to NOT rely on the `.oh-my-zsh` folder existing and instead touch another file. This is the paradigm we're moving towards for v8

Comment (m4olivei) [reply scripts/install-ohmyzsh.sh]: Ok cool. You mean like this for example right?

https://github.com/laravel/homestead/blob/master/scripts/install-mysql8.sh#L12

Comment (svpernova09) [reply scripts/install-ohmyzsh.sh]: Exactly!

**Comment by m4olivei** [comment]
There we go. Tested and working for me. I used the name `.ohmyzsh` for the flag file, which is slightly awkward b/c it sits beside `.oh-my-zsh` which is the source directory for oh my zsh. I think it's OK, but open to suggestions.

**Comment by svpernova09** [comment]
>  I used the name .ohmyzsh for the flag file, which is slightly awkward b/c it sits beside .oh-my-zsh which is the source directory for oh my zsh. I think it's OK, but open to suggestions.

Yeah, in v8 we shove these into another folder so it's less awkward.

**Comment by svpernova09** [comment]
Thanks!

**Pull request merged**